makefile

```diff
@@ -12,15 +12,35 @@ $(1)
 
 endef
 
+# Function to make sure that the git directory is clean.
+define check-git-status
+if [[ -n $$(git --no-pager status -s 2> /dev/null) ]] ;\
+then \
+	echo "Git tree is not clean. Did you forget to commit some files?" ;\
```

Comment:
This is also printed when generated files are not in sync, right? Perhaps mention that in the error message.